blackjax/inference/mh_step.py

```diff
@@ -0,0 +1,150 @@
+"""Metropolis-Hasting Step"""
```

Comment:
It is not really a MH step - probably more precise to call it "(Multinomial) sample from path" or "(Multinomial) sample from tree"

Reply:
I'll do a pass on naming/docstring once it's well tested and the API is fixed to make it as pedagogical as possible.